### ISSUE TITLE
Inconsistent indentation of Non-Latin `\item` and non-`\item` paragraphs (with linebreaks)

### ISSUE THREAD

**Issue by Andrew15-5** [cmhughes/latexindent.pl#431]
original .tex code
--

```latex
\begin{itemize}
  \item один;
  \item два;
  \item три три три три три три три три три три три три три три три три три три три;

    Другой текст.
  \item четыре.
\end{itemize}
```

yaml settings
--

```yaml
defaultIndent: "  "
modifyLineBreaks:
  textWrapOptions:
    blocksBeginWith:
      # other: 'три'
      other: '\w'
    columns: 81
    when: after
```

actual/given output
--

With `latexindent.pl file.tex`:

```latex
\begin{itemize}
  \item один;
  \item два;
  \item три три три три три три три три три три три три три три три три три три три;

        Другой текст.
  \item четыре.
\end{itemize}
```

With `latexindent.pl -m file.tex`:

```latex
\begin{itemize}
  \item один;
  \item два;
  \item три три три три три три три три три три три три три три три три три три
        три;

Другой текст.
  \item четыре.
\end{itemize}
```

desired or expected output
--

```latex
\begin{itemize}
  \item один;
  \item два;
  \item три три три три три три три три три три три три три три три три три три
        три;

        Другой текст.
  \item четыре.
\end{itemize}
```

anything else
--

Here I want to apply linebreaks as well as the correct indentation.

As we discussed earlier (https://github.com/cmhughes/latexindent.pl/issues/428#issuecomment-1458794462, https://github.com/cmhughes/latexindent.pl/issues/428#issuecomment-1458824151) I have to add some kind of regex to `modifyLineBreaks:textWrapOptions:blocksBeginWith:other` if I want to trigger linebreak for languages with non-Latin characters.

If I put (mostly or 100%) universal regex such as `\w` then it will work, but it will also affect paragraphs that are not with `\item` prefix but are still in the same environment as the other `\item` paragraphs. In the example paragraph, `Другой текст.` get a `0` indent level instead of `8`.

In other words, just applying the config file (without `-m` option) to the code is perfect, but additionally I need to apply linebreaks. If I instead add `-m` option then it will ruin the indentation but will correctly add linebreaks.

The only workaround (which I know about) is to change `other` option to `other: 'три'`. But this means that for every paragraph, I will have to modify the config file. And there is **no workaround** if `\item` paragraph and non-`\item` paragraph will start with the same word (i.e., `три`).

**Comment by cmhughes** [comment]
At the heart of this is the need to understand what the equivalent of a
capital letter (A, B, C,...) is in the language you're working in, and how
they map to regular expressions.

So, if you can answer that, hopefully we can progress.

**Comment by cmhughes** [comment]
In my language (English) the letters A, B, C, D,.... are represented
as`A-Z`  in regular expressions.

What is the equivalent in the language you're working in?

**Comment by Andrew15-5** [comment]
Ahhh! That's exactly what I did! Let me explain.

**Comment by Andrew15-5** [comment]
Now I at least know why you are talking about capital letters because I found something. After failing with `\w` I tested other regexes. Mainly, I tried all Latin and all Cyrillic letters separate and combined. And what is interesting to me is that Latin capital `L` change the behavior of formatter completely (2nd example under [`actual/given output`](https://github.com/cmhughes/latexindent.pl/issues/431#issue-1615378812)), while all other Latin letters did nothing.

I now understand that there is no reason to include `[a-zA-Z]` in the regex because English paragraphs already formatted correctly, therefore strange `L` also excluded (and the bug with it). After all that, I made a simple Cyrillic/Russian regex: `[а-яёА-ЯЁ]`. In many charsets including Unicode, Cyrillic letters `ё` and `Ё` are separate, therefore I had to include them too.

P.S. After seeing your last comment, I take back my words: `Now I at least know why you are talking about capital letters` because your intentions are completely different from "the `L` bug".

**Comment by cmhughes** [comment]
Great, so this is resolved?

**Comment by Andrew15-5** [comment]
So the working config is:

```yaml
defaultIndent: "  "
modifyLineBreaks:
  textWrapOptions:
    blocksBeginWith:
      other: '[а-яёА-ЯЁ]'
    columns: 81
    when: after
```

And here is the [Unicode table](https://www.rapidtables.com/code/text/unicode-characters.html) (to see where `ё` and `Ё` located relatively to the whole alphabet):

![image](https://user-images.githubusercontent.com/37143421/223809013-1dd4e8bc-92b6-4cca-9861-b28ada25ceb6.png)

But for languages like Japanese this is much harder because all characters are scattered across different ranges. But if I want to only add formatting to Hiragana, then range `[ぁ-ゖ]` probably would work (and changing `columns` to `41`):

![image](https://user-images.githubusercontent.com/37143421/223812405-881012b7-74ce-4294-a9d0-10a077a7a093.png)

**Comment by Andrew15-5** [comment]
> Great, so this is resolved?

Yes. I hope so... :)
